### DISCUSSION TITLE
Searching Through Tables

### DISCUSSION THREAD

**Discussion by ccigas** [Budibase/budibase#6463] [Help]
Hi, I am playing with Budi Base for the first time today and loving it so far. I am starting off with the client contact list template and just tweaking it a bit. I want to be able to search the default template but when I do add a search function to the table, I cant search everything in the table at once or not full strings. What I mean is this, if I am using that template, I choose Last Name for the filter on the search bar and have to put the full name in to actually search for it, and only the last name. (Method 2 from here https://docs.budibase.com/docs/adding-a-search-field). Is there a way to change that so I can search partly or all columns at once? 

I then tried method 1 in that link and besides needing multiple search bars which is fine but annoying, I then lose the View feature of the normal table. Is there a way to get that back at all? 

Thank you

**Comment by melohagan** [comment]
Hi @ccigas 

Thanks for trying Budibase!

You can try replacing the *Equals* filter with *Starts with*:
![Screenshot 2022-06-27 at 15 31 57](https://user-images.githubusercontent.com/101575380/175965909-9cc3218f-6697-4776-9a94-9a99e619d45e.png)

**Note** There appears to be a minor bug with this that caught me when I was trying, but this fix worked though: https://github.com/Budibase/budibase/issues/5549#issuecomment-1107441947

![Screenshot 2022-06-27 at 15 33 09](https://user-images.githubusercontent.com/101575380/175966172-67b71a3e-17de-4722-8610-700fa0d38a2e.png)

Unfortunately there is no way to 'OR' filters, so you can't partially filter on multiple columns yet. There's an issue open here: https://github.com/Budibase/budibase/issues/5993

**Reply by ccigas** [reply]
Thank you @melohagan ! This seems to work that way well, just hope for the other features to come out sometime soon. 

Would you know how to view the clients using method one in that link I mentioned in the op? Seems like its a solid option but not sure how I would view rows from there.

**Reply by notflip** [reply]
I got my search (using SQL) working using `%{{Form.Fields.Search}}%`, as you would in SQL in the filters
